src/stdlib_io.fypp

```diff
@@ -122,10 +129,17 @@ contains
       #:endif
 
       ! determine number or rows
-      nrow = number_of_rows(s)
+      nrow = number_of_rows(s) - skiprows_
+      if ( nrow < 0 ) call error_stop("loadtxt: skipping more rows than present.")
+      if ( max_rows_ < 0 .or. max_rows_ > nrow) max_rows_ = nrow
+
+      allocate(d(max_rows_, ncol))
+
+      do i = 1, skiprows_
+        read(s, *)
+      end do
```

Comment:
```suggestion
      do i = 1, skiprows_
        read(s, *, iostat = io)
        if (io .ne. 0) then
         allocate(d(0, 0))
         return
        end if
      end do
```
Or alternatively,
```suggestion
      do i = 1, skiprows_
        read(s, *, iostat = io, iomsg = msg)
        if (io .ne. 0) then
         error stop msg
        end if
      end do
```
@MuellerSeb Here are some possibilities. All are fine for me. The main question IMO is that if  the program should `error stop` or return a zero-size array when the file is "empty".

Reply:
Then I am still in favor of my suggested implementation or alternatively we could cut off `skip_rows_` at the number of present lines.

Reply:
Maybe some more thoughts: Catching this with `iostat`, is a bit strange, since we know beforehand what is going to happen, since the number of rows in the file was calculated. Putting an `error stop msg` there, could be a bit misleading, since we know exactly what the actual problem is: skipping more lines than present.
The other option to silently creating a 0-size array is the same result as cutting of `skiprows_` at the number of remaining rows. Then a 0-size array is also created with the current code.
If we vote against the error message I implemented, I would vote for cutting off `skip_rows_`.